.github/workflows/e2e.yml

```diff
@@ -0,0 +1,95 @@
+name: End2End tests
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  kind:
+    name: Kind
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Go
+        id: setup-go
+        uses: ./.github/actions/setup-go-cache
+        with:
+          cache-prefix: go-integration-kind
+
+      - name: Starting Armada cluster
+        run: make kind-all
```

Comment:
`make kind-all` installs the latest released Armada Operator Helm chart - https://github.com/armadaproject/armada-operator/blob/main/Makefile#L214
This way we run e2e tests for the latest released Operator, and in case of a change to Operator code or Helm chart, the e2e test won't cover that scenario.

I think it might be better to create a `make kind-all-local` or edit the `make kind-all-dev` to build the code and install from the local Helm chart.
That way we would cover the scenario to do an e2e test on current Operator code and on current Helm chart.

Reply:
Something like this?
https://github.com/armadaproject/armada-operator/pull/355/commits/561dced96666c2aaeb2019e7ae8ffa769b42bf35#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R218